pkg/master/controller.go

```diff
@@ -257,18 +260,20 @@ type EndpointReconciler interface {
 // masterCountEndpointReconciler reconciles endpoints based on a specified expected number of
 // masters. masterCountEndpointReconciler implements EndpointReconciler.
 type masterCountEndpointReconciler struct {
-	masterCount      int
-	endpointRegistry endpoint.Registry
+	masterCount       int
+	endpointRegistry  endpoint.Registry
+	configmapRegistry configmap.Registry
```

Comment:
also, read the configmap from disk, not directly from etcd. As we move flags into the configmap, the location of etcd will be one of them. Reading directly from etcd will make that impossible.